gnat2goto/driver/tree_walk.adb

```diff
@@ -1315,24 +1318,28 @@ package body Tree_Walk is
                       "Identifier not in class type. Type definition failed.");
          return;
       end if;
-      if not Is_Type (E)
-      then
+
+      if not Is_Type (E) then
          Report_Unhandled_Node_Empty (N, "Do_Full_Type_Declaration",
                                       "identifier not a type");
          return;
       end if;
 
       --  If this is the full_type_declaration of a previous
-      --  incomplete_type_declaration then the Incomplete_View of the
-      --  declaration will be present and the
+      --  private_type_declaration or incomplete_type_declaration then the
+      --  either it will have a previous private declaration or an
+      --  Incomplete_View of the declaration will be present and the
       --  full_type_declaration will have been registered when its
-      --  incomplete_type_declaration was processed.
-      --  If the Incomplete_View is not present then the
-      --   full_type_declaration has to be registered
-      if not (Present (Incomplete_View (N))) then
-         Put_Line ("We are going to do the dec");
-         Put_Line ("full_type_declaration with no incomplete_view");
-         Register_Type_Declaration (N, E);
+      --  private or incomplete_type_declaration was processed.
+      --  If it has no private declaration or the Incomplete_View is not
+      --  present then the full_type_declaration has to be registered
+      if not (Has_Private_Declaration (E)
+              or else Present (Incomplete_View (N)))
```

Comment:
According to the comment above this should be:
```
if not (Has_Private_Declaration (E)) or else not (Present (Incomplete_View (N)))
```

Reply:
The code is correct - if there is no private declaration or incomplete view then it is a straight forward full type declaration and it must be registered.  I think the comment was unclear.